### PR TITLE
Allow configuration of Prometheus log level (#2484)

### PR DESCRIPTION
**Pull request by liquidslr**
Solves #2484

Signed-off-by: Gaurav Kumar <gaurav.kumar9825@gmail.com>

### CODE REVIEW

**Comment by alpeb** [comment]
Hi @liquidslr , thanks for the contribution :+1: 
The flag need to be actually `--log.level={{.Values.ControllerLogLevel}}`, otherwise the linkerd-prometheus pod will refuse to start.

**Comment by liquidslr** [comment]
Thank you @alpeb, didn't notice that

File: chart/templates/prometheus.yaml [hunk 1]
```diff
@@ -97,7 +97,7 @@ spec:
         - "--storage.tsdb.path=/{{.Values.PrometheusVolumeName}}"
         - "--storage.tsdb.retention=6h"
         - "--config.file=/etc/prometheus/prometheus.yml"
-        - "-log-level={{.Values.ControllerLogLevel}}"
+        - "--log-level={{.Values.ControllerLogLevel}}"
```
[the remark this answers is not in the document]
Comment (liquidslr) [reply chart/templates/prometheus.yaml]: Sorry for the repetitive mistakes, did a full build of the control plane and check, everything seems to be working fine.

[the remark this answers is not in the document]
Comment (alpeb) [reply chart/templates/prometheus.yaml]: No worries :slightly_smiling_face: 
One thing left is to handle the `panic` and `fatal` values as explained above. One option would be to create a new template var `{{.Values.PrometheusLogLevel}}` that coalesces `panic` and `fatal` into `error`. Something like this:
```diff
diff --git a/chart/templates/prometheus.yaml b/chart/templates/prometheus.yaml
index 26dd807c..34240255 100644
--- a/chart/templates/prometheus.yaml
+++ b/chart/templates/prometheus.yaml
@@ -97,7 +97,7 @@ spec:
         - "--storage.tsdb.path=/{{.Values.PrometheusVolumeName}}"
         - "--storage.tsdb.retention=6h"
         - "--config.file=/etc/prometheus/prometheus.yml"
-        - "--log.level={{.Values.ControllerLogLevel}}"
+        - "--log.level={{.Values.PrometheusLogLevel}}"
         readinessProbe:
           httpGet:
             path: /-/ready
diff --git a/cli/cmd/install.go b/cli/cmd/install.go
index aae77e17..758989cf 100644
--- a/cli/cmd/install.go
+++ b/cli/cmd/install.go
@@ -35,6 +35,7 @@ type installConfig struct {
        UUID                        string
        CliVersion                  string
        ControllerLogLevel          string
+       PrometheusLogLevel          string
        ControllerComponentLabel    string
        CreatedByAnnotation         string
        EnableTLS                   bool
@@ -153,6 +154,11 @@ func validateAndBuildConfig(options *installOptions) (*installConfig, error) {
                return nil, err
        }
 
+       prometheusLogLevel := options.controllerLogLevel
+       if prometheusLogLevel == "panic" || prometheusLogLevel == "fatal" {
+               prometheusLogLevel = "error"
+       }
+
        return &installConfig{
                Namespace:                   controlPlaneNamespace,
                ControllerImage:             fmt.Sprintf("%s/controller:%s", options.dockerRegistry, options.linkerdVersion),
@@ -166,6 +172,7 @@ func validateAndBuildConfig(options *installOptions) (*installConfig, error) {
                UUID:                        uuid.NewV4().String(),
                CliVersion:                  k8s.CreatedByAnnotationValue(),
                ControllerLogLevel:          options.controllerLogLevel,
+               PrometheusLogLevel:          prometheusLogLevel,
                ControllerComponentLabel:    k8s.ControllerComponentLabel,
                ControllerUID:               options.controllerUID,
                CreatedByAnnotation:         k8s.CreatedByAnnotation,
```
@siggy what do you think?

[the remark this answers is not in the document]
Comment (siggy) [reply chart/templates/prometheus.yaml]: @alpeb This looks great.

@liquidslr Almost there, I think if you apply this change, fix up some unit and integration tests, and rebase, you'll be good to go.

```bash
bin/dep ensure
go test ./... # unit tests

DOCKER_TRACE=1 bin/docker-build-proxy
bin/linkerd install --controller-log-level fatal | kubectl apply -f -
bin/linkerd check --expected-version=$(bin/root-tag)

bin/test-run `pwd`/bin/linkerd # integration tests
```

**Comment by liquidslr** [comment]
@klingerf  can you please review now, i am unable to figure out what else am i missing

**Comment by liquidslr** [comment]
Thanks, @alpeb @siggy for all the help <3

**Pull request merged**